integration_test/tests/advertisingTest.ts

```diff
@@ -0,0 +1,181 @@
+import { TestScope } from 'cavy';
+import {
+  callPlayer,
+  callPlayerAndExpectEvent,
+  callPlayerAndExpectEvents,
+  EventSequence,
+  EventType,
+  expectEvent,
+  expectEvents,
+  loadSourceConfig,
+  RepeatedEvent,
+  startPlayerTest,
+} from '../playertesting';
+import { Sources } from './helper/Sources';
+import {
+  AdItem,
+  AdSourceType,
+  AdvertisingConfig,
+} from 'bitmovin-player-react-native';
+import { AdTags } from './helper/Ads';
+import { Platform } from 'react-native';
+
+export default (spec: TestScope) => {
+  function commonAdvertisingTests(
+    adItems: AdItem[],
+    type: 'VAST' | 'VMAP' | 'progressive'
+  ) {
+    spec.describe(`scheduling ${type} ads via AdvertisingConfig`, () => {
+      spec.it('emits AdScheduled events', async () => {
+        const advertisingConfig = {
+          schedule: adItems,
+        } as AdvertisingConfig;
+        await startPlayerTest({ advertisingConfig }, async () => {
+          await callPlayerAndExpectEvents((player) => {
+            player.load(Sources.artOfMotionHls);
+          }, RepeatedEvent(EventType.AdScheduled, adItems.length));
+        });
+      });
+    });
+    spec.describe(`scheduling ${type} ads via scheduleAd API`, () => {
+      spec.it('emits AdScheduled events', async () => {
+        await startPlayerTest({}, async () => {
+          await loadSourceConfig(Sources.artOfMotionHls);
+          await callPlayerAndExpectEvents((player) => {
+            adItems.forEach((adItem) => player.scheduleAd(adItem));
+          }, RepeatedEvent(EventType.AdScheduled, adItems.length));
+        });
+      });
+    });
+    spec.describe(`playing ${type} ads`, () => {
+      spec.it('emits Ad events', async () => {
+        const advertisingConfig = {
+          schedule: adItems,
+        } as AdvertisingConfig;
+        await startPlayerTest({ advertisingConfig }, async () => {
+          await callPlayer(async (player) => {
+            player.load(Sources.artOfMotionHls);
+            player.play();
+          });
+          if (type !== 'progressive') {
+            await expectEvent(EventType.AdManifestLoaded, 20);
+          }
+          await expectEvents(
+            EventSequence(
+              EventType.AdBreakStarted,
+              EventType.AdStarted,
+              EventType.AdQuartile,
+              EventType.AdQuartile,
+              EventType.AdQuartile,
+              EventType.AdFinished,
+              EventType.AdBreakFinished
+            ),
+            18
```

Comment:
Is there a reason for this quite specific 18? Just a though, what if we increase the default timeout to account for Android emulators being a bit slower and set it to 20?

Reply:
I would not touch the default timeout as usually 10 seconds is pretty long already.
This is specific for ads on the Android emulator so belongs to the specific test.